mmdeploy/codebase/mmdet3d/deploy/voxel_detection.py

```diff
@@ -1,21 +1,54 @@
 # Copyright (c) OpenMMLab. All rights reserved.
+from copy import deepcopy
 from typing import Any, Dict, List, Optional, Sequence, Tuple, Union
 
-import mmcv
 import mmengine
 import numpy as np
 import torch
 import torch.nn as nn
-from mmcv.parallel import collate, scatter
-from mmdet3d.core.bbox import get_box_type
-from mmdet3d.datasets.pipelines import Compose
+from mmdet3d.structures import get_box_type
+from mmengine import Config
+from mmengine.dataset import Compose, pseudo_collate
+from mmengine.model import BaseDataPreprocessor
+from mmengine.registry import Registry
 from torch.utils.data import DataLoader, Dataset
 
-from mmdeploy.codebase.base import BaseTask
-from mmdeploy.codebase.mmdet3d.deploy.mmdetection3d import MMDET3D_TASK
-from mmdeploy.utils import Task, get_root_logger, load_config
+from mmdeploy.codebase.base import CODEBASE, BaseTask, MMCodebase
+from mmdeploy.utils import (Backend, Codebase, Task, get_backend,
+                            get_root_logger)
 from .voxel_detection_model import VoxelDetectionModel
 
+MMDET3D_TASK = Registry('mmdet3d_tasks')
+
+
+@CODEBASE.register_module(Codebase.MMDET3D.value)
+class MMDetection3d(MMCodebase):
+    """MMDetection3d codebase class."""
+
+    task_registry = MMDET3D_TASK
+
+
+def _get_dataset_metainfo(model_cfg: Config):
+    """Get metainfo of dataset.
+
+    Args:
+        model_cfg Config: Input model Config object.
+
+    Returns:
+        list[str]: A list of string specifying names of different class.
+    """
+
+    for dataloader_name in [
+            'test_dataloader', 'val_dataloader', 'train_dataloader'
+    ]:
+        if dataloader_name not in model_cfg:
+            continue
+        dataloader_cfg = model_cfg[dataloader_name]
+        dataset_cfg = dataloader_cfg.dataset
+        if 'metainfo' in dataset_cfg:
+            return dataset_cfg.metainfo
+    return None
+
```

Comment:
Could add `register_all_modules(cls)` like other derivate of MMCodbase to load modules.

Reply:
done.